src/main/scala/uk/gov/nationalarchives/omega/api/repository/SparqlParams.scala

```diff
@@ -77,7 +77,9 @@ object SparqlParams extends AgentTypeMapper {
   private def getDateTimeMap(listAgentSummary: ListAgentSummary): Map[String, String] =
     listAgentSummary.versionTimestamp match {
       case Some("all") | Some("latest") | None => Map.empty
-      case Some(dateTimeValue) =>
+      case Some(dateTimeValue)                 =>
+        // TODO(RW) we should attempt to parse this date to ensure it is valid (see PACT-1087)
```

Comment:
With your comment in mind, would it perhaps make sense to not treat xs:dateTime as a String type internally but instead treat it as XMLGregorianCaldendar or other type?

Reply:
I think the map can be of type `Map[String, XMLGregorianCalendar]` - and I have changed this - but the `ListAgentSummary.versionTimestamp` can also be "all" or "latest" so I can't use a date type in there.